### DISCUSSION TITLE
QST: GeoSeries.intersection when both polygons cross antimeridian produces an EMPTY polygon

### DISCUSSION THREAD

**Discussion by antdea** [geopandas/geopandas#2758] [Q&A] [answered]
#### Code Sample, a copy-pastable example

```python
from shapely.geometry import Point
import geopandas

pt1 = Point(175,0)
pt2 = Point(-175,0)

gs1 = geopandas.GeoSeries([pt1], crs=4326)
gs1 = gs1.to_crs(4087)
gs1 = gs1.buffer(2000000)

gs2 = geopandas.GeoSeries([pt2], crs=4326)
gs2 = gs2.to_crs(4087)
gs2 = gs2.buffer(2000000)

intersection = gs1.intersection(gs2)
print(intersection)

```

#### Problem description

When both polygons cross the dateline then any intersection of the 2 polygons produces and EMPTY polygon. This does not happen when only 1 polygon crosses the dateline.

#### Expected Output

A polygon showing the intersection should be created.

<details>

SYSTEM INFO
-----------
python     : 3.11.0 (v3.11.0:deaf509e8f, Oct 24 2022, 14:43:23) [Clang 13.0.0 (clang-1300.0.29.30)]
executable : /Users/anthonydeakin/Library/Caches/pypoetry/virtualenvs/shapely-test-02NeaYvN-py3.11/bin/python3
machine    : macOS-12.5-arm64-arm-64bit

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : 3.11.1
GEOS lib   : None
GDAL       : None
GDAL data dir: None
PROJ       : 9.1.0
PROJ data dir: /Users/anthonydeakin/Library/Caches/pypoetry/virtualenvs/shapely-test-02NeaYvN-py3.11/lib/python3.11/site-packages/pyproj/proj_dir/share/proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 0.12.2
numpy      : 1.24.1
pandas     : 1.5.2
pyproj     : 3.4.1
shapely    : 2.0.0
fiona      : None
geoalchemy2: None
geopy      : None
matplotlib : None
mapclassify: None
pygeos     : None
pyogrio    : None
psycopg2   : None
pyarrow    : None
rtree      : None

</details>

**Comment by brendan-ward** [comment, marked as answer]
This isn't a bug, I think it is a misunderstanding of the coordinates.  They don't overlap based on the coordinates you've provided:

<img width="595" alt="image" src="https://user-images.githubusercontent.com/3375604/212348895-3d1f0cee-f321-4b1f-b998-996c2ecc8528.png">

The buffer of your first point extends from the antimeridian right (greater than 180) and your second point extends from the antimeridian left (smaller values than -180).

To get around this, you need to put them into overlapping coordinate space; this is not done automatically for you.

There are a couple ways you could do this:
1. split the buffers at the antimeridian using [shapely.clip_by_rect](https://shapely.readthedocs.io/en/stable/manual.html#shapely.ops.clip_by_rect), then move the part that extends beyond the equivalent of longitude [-180 / 180] in `EPSG:4087` to the opposite side of the coordinate space using [shapely.transform](https://shapely.readthedocs.io/en/stable/manual.html#shapely.ops.transform), then do the overlay.

2. pick one of the geometries to move to the other side of the coordinate space using `shapely.transform`

So for instance you could pick the one on the right side of the coordinate space, and subtract the equivalent of longitude 360 in `EPSG:4087` (i.e., the width of the X domain) from all x coordinates to move it to the left side, something like  `shapely.transform(gs1.geometry.values, lambda g: g - [x_domain_width, 0])[0]` (taking first index since this is a array-based operation)

Similar discussion in shapely at: https://github.com/shapely/shapely/discussions/1716
